### ISSUE TITLE
Feature Request: Support TypeScript conditional types

### ISSUE THREAD

**Issue by Cryrivers** [prettier/prettier#4006]
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACGATwzjwBE44MAlOAQwBNoAbQgHgBUA+PAXgB0oeYXg5508KAwDOeOlEIBtALp4A-OUo16TKKwCC2bHXYdFUAK4BbAEZxsynkkEjR4tJJl4INgFZwwfA0KKlpGFkIAeT8AmE4nFxEOAG4QABoQCAwYAEtoaWRQOiMIAHcABWKEApQ6ADcIHIZ0kDppGGQAMzpmaTgMm2MwAGs4GABlDDowHKgAc2QYbAt+kCYwLp6+jNm+3HLjOas6Td7V32k0ACEh0Ym6KzgAGVm4U+3MixgML4Amd9WU2we2QICIJGkYGwOWyLQw0NgAHUmjAABbIAAcAAYMvCIH1EcYMKD4XA9nU3hlsHAAI4WHLUg50I4nJDdM4ZPpWHKLZaraSzObMOAARQsEHgAIyMDoNmRDDRyF+0uMOWYgoAwhArMdQVBoJSQBY+hxZTV2X0AL6WoA)
```sh
--parser typescript
```

**Input:**
```tsx
export type DeepReadonly<T> =
    T extends any[] ? DeepReadonlyArray<T[number]> :
    T extends object ? DeepReadonlyObject<T> :
    T;
```

**Output:**
```tsx
SyntaxError: ';' expected. (2:7)
  1 | export type DeepReadonly<T> =
> 2 |     T extends any[] ? DeepReadonlyArray<T[number]> :
    |       ^
  3 |     T extends object ? DeepReadonlyObject<T> :
  4 |     T;
```

**Expected behavior:**
```tsx
export type DeepReadonly<T> = T extends any[]
  ? DeepReadonlyArray<T[number]>
  : T extends object ? DeepReadonlyObject<T> : T;
```

**Comment by utatti** [comment]
I guess there is still this issue with 1.11. The playground above emits the same error as 1.10.2.

**Comment by suchipi** [comment]
This is fixed in 1.11.1.
